### PR TITLE
[test]: test with py39 + torch 1.8 nightly

### PR DESCRIPTION
**Pull request by min-xu-ai**
# Before submitting

- [ ] Was this discussed/approved via a Github issue? (no need for typos, doc improvements)
- [ ] Did you read the [contributor guideline](https://github.com/facebookresearch/fairscale/blob/master/CONTRIBUTING.md)?
- [ ] Did you make sure to update the docs?
- [ ] Did you write any new necessary tests?

## What does this PR do?
Fixes # (issue).

## PR review
Anyone in the community is free to review the PR once the tests have passed.
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

**Comment by min-xu-ai** [comment]
This should make our test coverage better with py39. The torch-pg dependency can be updated later.

**Comment by blefaudeux** [comment]
we could remove torchvision from requirements-test Min, and only install it on the agent which runs the benchmark, we only run it once anyway ?

**Comment by min-xu-ai** [comment]
> 
> 
> we could remove torchvision from requirements-test Min, and only install it on the agent which runs the benchmark, we only run it once anyway ?

Great idea. I just did that. I separated the requirements for benchmarks out. See this commit.

https://github.com/facebookresearch/fairscale/pull/339/commits/56563b6987ed09471bf07621b410a1b36f32226f

File: requirements-benchmarks.txt [hunk 1]
```diff
@@ -0,0 +1,7 @@
+# Bring in everything that tests depends on.
```
Comment (blefaudeux) [inline requirements-benchmarks.txt:1]: _like_

File: fairscale/optim/adascale.py [hunk 2]
```diff
@@ -514,8 +514,9 @@ def add_param_group(self, pg: Dict) -> None:
         # Extend the states.
         for name in self._state.keys():
             assert name.startswith("grad_sqr_avg") or name.startswith("grad_var_avg"), name
-            if isinstance(self._state[name], int):
-                # This is the "_count" variable.
+            if name.endswith("_count"):
```
Comment (blefaudeux) [inline fairscale/optim/adascale.py:517]: hmm, is that expected ? looks like a logic change

**Review by blefaudeux — APPROVED** [review]
Lokos like there are a few adascale changes which slipped in, checking here that you're on top of that ? Else LGTM !

File: fairscale/optim/adascale.py (hunk 2 above)
Comment (min-xu-ai) [reply fairscale/optim/adascale.py:517]: right, it is induced by mypy checking. newer numpy version actually enabled mypy to find out that adascale code here had a mixed typing, so it flagged it. I fixed it by change the typing to not be mixed. So, it is a typing change. No logic changes.

**Pull request merged**